tests/components/onkyo/test_config_flow.py

```diff
@@ -455,12 +410,13 @@ async def test_import_fail(
     error: str,
 ) -> None:
     """Test import flow failed."""
-    with (
-        patch(
-            "homeassistant.components.onkyo.config_flow.async_interview",
-            return_value=None,
-            side_effect=exception,
-        ),
+
+    async def mock_discover(host, discovery_callback, timeout):
+        raise exception
+
+    with patch(
+        "homeassistant.components.onkyo.receiver.pyeiscp.Connection.discover",
+        new=mock_discover,
```

Comment:
```suggestion
        side_effect=exception,
```

Why not use side_effect ?